RFCs/0070-Fn::Size.md

```diff
@@ -0,0 +1,104 @@
+# RFC for Fn::Size
+
+* **Original Author(s):**: @juegong2
+* **Tracking Issue**: https://github.com/aws-cloudformation/cfn-language-discussion/issues/70
+
+# Summary
+
+We will support an intrinsic function called `Fn::Size` that will return the number of elements in a given list.
+
+# Motivation
+
+CloudFormation users may want to get the number of elements in a given list to set certain resource parameters or generate conditions. It may also help to check if index is out of bound for !Select function. This RFC proposes to support the use case of getting the number of elements of a given list.
+
+* Reference:
+  * https://github.com/aws-cloudformation/cfn-language-discussion/issues/61
+
+# Example(s)
+
+Use `Ref` on a Parameter of type `CommaDelimitedList` or `List<Number>`
+
+```json
+{
+  "Parameters": {
+    "InstanceTypes": {
+      "Type": "CommaDelimitedList"
+    },
+    "NumberList": {
+      "Type": "List<Number>"
+    }
+  }
+  "Conditions": {
+    "Has3InstanceTypes": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": {"Ref": "InstanceTypes"}
+        },
+        3
+      ]
+    },
+    "Has2NumbersInList": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": {"Ref": "NumberList"}
+        },
+        2
+      ]
+    }
+  }
+}
+```
+
+Use a hardcoded list
+```json
+{
+  "Conditions": {
+    "Has3InstanceTypes": {
+      "Fn::Equals": [
+        {
+          "Fn::Size": [ "m5.large", "m5.xlarge" ]
+        },
+        3
+      ]
+    }
+  }
+}
+```
+
+# Limitation
```

Comment:
Thanks for making the wording more precise here. I suspect we'll get feedback on this as from a customer point of view, this is not ideal (ideally should support this case as well). Anticipating that, maybe we should add something like this to the bottom of this section: "Note: This is a short-term limitation due to underlying implementation constraints. In the fullness of time, this limitation should be removed."